tests/components/updater/test_init.py

```diff
@@ -44,56 +45,118 @@ def mock_get_uuid():
         yield mock
 
 
+@pytest.fixture
+def mock_utcnow():
+    """Fixture to mock utcnow."""
+    with patch("homeassistant.components.updater.dt_util.utcnow") as mock:
+        yield mock
+
+
 @asyncio.coroutine
-def test_new_version_shows_entity_after_hour(
-    hass, mock_get_uuid, mock_get_newest_version
-):
-    """Test if new entity is created if new version is available."""
+def test_new_version_shows_entity_startup(hass, mock_get_uuid, mock_get_newest_version):
+    """Test if binary sensor is unavailable at first."""
+    mock_get_uuid.return_value = MOCK_HUUID
+    mock_get_newest_version.return_value = mock_coro((NEW_VERSION, RELEASE_NOTES))
+
+    res = yield from async_setup_component(hass, updater.DOMAIN, {updater.DOMAIN: {}})
+    assert res, "Updater failed to set up"
+
+    yield from hass.async_block_till_done()
+
+    assert hass.states.is_state("binary_sensor.updater", "unavailable")
+    assert "newest_version" not in hass.states.get("binary_sensor.updater").attributes
+    assert "release_notes" not in hass.states.get("binary_sensor.updater").attributes
+
+    entity_registry = yield from hass.helpers.entity_registry.async_get_registry()
+    entity_registry.async_update_entity(
+        "binary_sensor.updater", new_entity_id="binary_sensor.new_entity_id"
+    )
+
+    yield from hass.async_block_till_done()
+
+    assert hass.states.is_state("binary_sensor.new_entity_id", "unavailable")
+    assert hass.states.is_state("binary_sensor.updater", "unavailable") is False
+
```

Comment:
Trigger an update here, by moving time forward, and assert that the new entity updates the state correctly while the old entity should still not have a state. That way we check that the dispatcher has been unsubscribed and resubscribed correctly.

Reply:
Done. From my point of view, you can merge this if you want to.